tests/Integration/Behaviour/Features/Context/CurrencyFeatureContext.php

```diff
@@ -121,4 +122,121 @@ public function checkCurrencyWithNameExists($currencyName)
     {
         $this->checkFixtureExists($this->currencies, 'Currency', $currencyName);
     }
+
+    /**
+     * @Then currency :reference should be :isoCode
+     */
+    public function assertCurrencyIsoCode($reference, $isoCode)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        if ($currency->iso_code !== $isoCode) {
+            throw new RuntimeException(sprintf(
+                'Currency "%s" has "%s" iso code, but "%s" was expected.',
+                $reference,
+                $currency->iso_code,
+                $isoCode
+            ));
+        }
+    }
+
+    /**
+     * @Then /^currency "(.*)" should have status (enabled|disabled)$/
+     */
+    public function assertCurrencyStatus($reference, $status)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+        $expectedStatus = $status === 'enabled';
+
+        if ($currency->active != $expectedStatus) {
+            throw new RuntimeException(sprintf(
+                'Currency "%s" has status "%s", but "%s" was expected.',
+                $reference,
+                $currency->active,
+                $expectedStatus
+            ));
+        }
+    }
+
+    /**
+     * @Then currency :reference exchange rate should be :exchangeRate
+     */
+    public function assertCurrencyExchangeRate($reference, $exchangeRate)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        if ((float) $currency->conversion_rate != (float) $exchangeRate) {
+            throw new RuntimeException(sprintf(
+                'Currency "%s" has "%s" exchange rate, but "%s" was expected.',
+                $reference,
+                $currency->conversion_rate,
+                $exchangeRate
+            ));
+        }
+    }
+
+    /**
+     * @Then currency :currencyReference should be available in shop :shopReference
+     */
+    public function assertCurrencyIsAvailableInShop($currencyReference, $shopReference)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($currencyReference);
+        /** @var \Shop $shop */
+        $shop = SharedStorage::getStorage()->get($shopReference);
+
+        if (!in_array($shop->id, $currency->getAssociatedShops())) {
```

Comment:
```suggestion
        if (!in_array($shop->id, $currency->getAssociatedShops(), true)) {
```

Reply:
Otherwise if by magic you have 0 in your shop id, it will always return true :/

Reply:
Are you sure? Can you give code example? I'm not sure if I get it. :/

By the way, ` $currency->getAssociatedShops()` returns array of strings `['1', '2']` instead of integers.

Reply:
My bad, it happens when you have false, or empty string in the return value.
Because of `==` comparaison instead of `===`

`var_dump(in_array('0', [false, ''])); # => true`